src/stan/model/indexing/lvalue.hpp

```diff
@@ -759,12 +731,12 @@ inline void assign(
     Mat1&& x,
     const cons_index_list<Idx, cons_index_list<index_omni, nil_index_list>>&
         idxs,
-    const Mat2& y, const char* name = "ANON", int depth = 0) {
-  assign(x, index_list(idxs.head_), y, name, depth + 1);
+    Mat2&& y, const char* name = "ANON", int depth = 0) {
+  assign(x, index_list(idxs.head_), std::forward<Mat2>(y), name, depth + 1);
```

Comment:
No benefit from PF here either.

Reply:
There is one edge case where pf would go off here (`index_omni, index_omni`)

Reply:
Oh, right.

Reply:
Eiter make the argument `const Mat2& y` or put `forward` call back in.

Reply:
Word I added the forward!